setup.sh

```diff
@@ -0,0 +1,189 @@
+      FA_TENANT_ID="${1:-}"
+      shift
+      ;;
+    *) argerr "Unsupported flag '${1}'" ;;
+    esac
+  done
+
+  echo ""
+
+  if [[ -z ${FA_URL:-} ]]; then
+    echo "⛔️ missing required --url param"
+    show_help
+    exit 1
+  else
+    if ! [[ ${FA_URL} == "http"* ]]; then
+      echo "⛔️ URL must be in the format https://your.fusionauth.host"
+      exit 1
+    fi
+    echo "✅ URL = ${FA_URL}"
+  fi
+
+  if [[ -z ${FA_API_KEY:-} ]]; then
+    echo "⛔️ missing required --key param"
+    show_help
+    exit 1
+  else
+    echo "✅ API Key = ${FA_API_KEY}"
+  fi
+
+  if [[ -z ${FA_TENANT_ID:-} ]]; then
+    FA_TENANT_ID="efb21cfc-fa60-46f4-9598-889151e58517"
+    echo "⚠️ --tenant was not set, using default value"
+  fi
+  echo "✅ Tenant ID = ${FA_TENANT_ID}"
+}
+
+function set_json_parser() {
+  # Try to find a JSON parser.
+  if command -v jq &>/dev/null; then
+    PARSER=jq
+  elif command -v python3 &>/dev/null; then
+    PARSER="python3 -m json.tool -"
+  elif command -v json_pp &>/dev/null; then
+    PARSER=json_pp
+  else
+    PARSER="cat -"
+  fi
+
+  if [[ ${PARSER} == "cat -" ]]; then
+    echo "⚠️ Could not find a JSON parser. JSON output will be unformatted."
+  else
+    echo "✅ JSON parser = ${PARSER}"
+  fi
+}
+
+function check_connection() {
+  echo -e "\nChecking connection"
+
+  RESPONSE=$(curl -X GET -s -o /dev/null -w "%{http_code}" \
+    -H "Content-Type: application/json" \
+    -H "Authorization: ${FA_API_KEY}" \
+    -H "X-FusionAuth-TenantId: ${FA_TENANT_ID}" \
+    -H "Cache-Control: no-cache" \
+    "${FA_URL}/api/tenant/${FA_TENANT_ID}")
+
+  if [[ ${RESPONSE} == "200" ]]; then
+    echo "✅ OK!"
+  else
+    echo "⛔️ Got a ${RESPONSE} response code"
+    echo "  - Check that the API key is correct, and has necessary permissions"
+    echo "  - Check that the tenant ID for the 'Default' tenant is correct"
+    exit 1
+  fi
+}
+
+function create_application() {
+  echo -e "\nCreating Application 'FusionAuthLoadTesting'"
+
+  curl -s -S -X POST \
+    -H "Content-Type: application/json" \
+    -H "Authorization: ${FA_API_KEY}" \
+    -H "X-FusionAuth-TenantId: ${FA_TENANT_ID}" \
+    -H "Cache-Control: no-cache" \
+    -d \
+    '{
+     "application": {
+       "name": "FusionAuthLoadTesting",
+       "oauthConfiguration": {
+          "authorizedRedirectURLs": ["https://acme.com/redirect"],
+          "clientSecret": "ZDA5Yzk0NTUtYzg4ZS00ZmNhLThmOGUtZDFkN2M3YTllNDY1",
+          "enabledGrants": ["authorization_code"]
+       },
+       "roles": ["admin", "user"]
+     }
+  }' "${FA_URL}/api/application/11e7ea7b-784d-4687-bf2d-4f8ee479a4dd" | $PARSER
+}
+
+function update_tests() {
+  echo -e "\nUpdating test files..."
+
+  TESTS_DIR=build/dist
+
+  echo "- User Registrations"
+  sed -i -e "s#https://local.fusionauth.io#${FA_URL}#g" "${TESTS_DIR}/User-Registrations.json"
+  sed -i -e "s#bf69486b-4733-4470-a592-f1bfce7af580#${FA_API_KEY}#g" "${TESTS_DIR}/User-Registrations.json"
```

Comment:
I think this should be the default value instead of always being the replace target.

If we don't want that assumption then it would probably make more sense to put a <YOUR KEY HERE> in place of that in the source json files

Reply:
Same deal as tenant ID, I can set local as the default.